app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -0,0 +1,544 @@
+              R.string.mobile_data_message_confirmation)
+      );
+      builder.setCancelable(false);
+      AlertDialog dialog = builder.create();
+      dialog.show();
+    }
+  }
+
+  @Override public void serverStarted(@NonNull String ip) {
+    this.ip = ip;
+    serverTextView.setText(getString(R.string.server_started_message, this.ip));
+    startServerButton.setText(getString(R.string.stop_server_label));
+    startServerButton.setBackgroundColor(getResources().getColor(R.color.stopServer));
+    isServerStarted = true;
+  }
+
+  @Override public void serverStopped() {
+    serverTextView.setText(getString(R.string.server_textview_default_message));
+    startServerButton.setText(getString(R.string.start_server_label));
+    startServerButton.setBackgroundColor(getResources().getColor(R.color.greenTick));
+    isServerStarted = false;
+  }
+
+  @Override public void serverFailed() {
+    Toast.makeText(this, R.string.server_failed_message, Toast.LENGTH_LONG).show();
+  }
+
+  @Override public void hotspotTurnedOn(@NonNull WifiConfiguration wifiConfiguration) {
+
+    //Show an alert dialog for hotspot details
+    AlertDialog.Builder builder = new AlertDialog.Builder(this, dialogStyle());
+
+    ProgressDialog progressDialog =
+        ProgressDialog.show(this, getString(R.string.progress_dialog_starting_server), "",
+            true);
+    progressDialog.show();
+
+    builder.setPositiveButton(android.R.string.ok, (dialog, id) -> {
+      final Handler handler = new Handler();
+      handler.postDelayed(() -> {
+        progressDialog.dismiss();
+        startService(ACTION_START_SERVER);
+      }, 2000);
+    });
+
+    builder.setTitle(this.getString(R.string.hotspot_turned_on));
+    builder.setMessage(
+        this.getString(R.string.hotspot_details_message) + "\n" + this.getString(
+            R.string.hotspot_ssid_label) + " " + wifiConfiguration.SSID + "\n" + this.getString(
+            R.string.hotspot_pass_label) + " " + wifiConfiguration.preSharedKey);
+
+    builder.setCancelable(false);
+    AlertDialog dialog = builder.create();
+    dialog.show();
+
+    //setupServer();
+  }
+
+  void setupWifiSettingsIntent() {
+    final Intent intent = new Intent(Intent.ACTION_MAIN, null);
+    intent.addCategory(Intent.CATEGORY_LAUNCHER);
+    final ComponentName cn =
+        new ComponentName("com.android.settings", "com.android.settings.TetherSettings");
+    intent.setComponent(cn);
+    intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+    startActivity(intent);
+  }
+
+  @Override public void hotspotFailed() {
+    //Show a dialog to turn off default hotspot
+
+    alertDialogShower.show(KiwixDialog.HotspotFailed.INSTANCE,
+        new Function0<Unit>() {
+          @Override public Unit invoke() {
+            setupWifiSettingsIntent();
+            return Unit.INSTANCE;
+          }
+        });
+  }
+
+  @Override public void hotspotState(@Nullable Boolean state) {
```

Comment:
`state` tells me absolutely nothing, what does `state is true` mean compared to `state is false`?
I can guess that true means it is active/enabled/started but it means you had to add these comments to explain it.
`Boolean` you also don't need this boxed type I would imagine? use primitives where possible as they can't be null.

Also should the activity be the one doing this? Where is this state coming from again?

Reply:
Fixed. 3ee5e7cde295993d7809ec82824e4f1cd6b656bd